src/lazy/text/buffer.rs

```diff
@@ -1597,11 +1671,11 @@ mod tests {
     }
 
     impl MatchTest {
-        /// Takes an `input` string and appends a trailing space to it, guaranteeing that the
+        /// Takes an `input` string and appends a trailing value to it, guaranteeing that the
         /// contents of the input are considered a complete token.
         fn new(input: &str) -> Self {
             MatchTest {
-                input: format!("{input} "), // add trailing space
+                input: format!("{input}\n0"), // add whitespace and a trailing value
```

Comment:
🗺️ Long string syntax is very greedy. It's impossible to declare a long string "complete" until you've run into something that _isn't_ a long string. Later on I'll add support for declaring that the input stream is done, but for the moment this workaround allows me to use the same test infrastructure on long strings.